pex/variables.py

```diff
@@ -668,45 +672,118 @@ def unzip_dir(
 
 
 def venv_dir(
```

Comment:
So, this consumes the constraints, the pex_path, and the interpreter rather than actually executing interpreter selection first, right? Is that for performance reasons? Because it seems like if you ran interpreter selection first, then the venv would always be keyed by the final interpreter, and the extra parameters here wouldn't be necessary.

Reply:
It is for performance reasons. Even importing `PythonInterpreter` is a ~20ms perf hit and we're talking re-exec  times of 50ms.